File: solutions/r/sieve/1/sieve.R
sieve <- function(limit) {
  primes <- c()
  if(limit >= 2) {
    append(primes,2)
    for(i in 2:limit) {
      is_prime <- TRUE
      limit_i <- sqrt(i)
      for(j in primes) {
        if(j > limit_i) {
          break
        }
        if(i%%j == 0) {
          is_prime <- FALSE
          break
        }
      }
      if(is_prime) {
        primes <- c(primes,i)
      }
    }
  }
  return(primes)
}

File: solutions/julia/sieve/1/sieve.jl
function sieve(limit)
    if limit <= 1 return [] end
    new = [2]
    if limit >= 3
        test = 3
        while test <= limit
            pass = true
            for prime in new
                if prime*prime > test break end
                if test%prime == 0
                    pass = false
                    break
                end
            end
            if pass push!(new,test) end
            test += 2
        end
    end
    return new
end
